common/src/main/java/org/conscrypt/HpkeImpl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+package org.conscrypt;
+
+import static org.conscrypt.HpkeSuite.AEAD_AES_128_GCM;
+import static org.conscrypt.HpkeSuite.AEAD_AES_256_GCM;
+import static org.conscrypt.HpkeSuite.AEAD_CHACHA20POLY1305;
+import static org.conscrypt.HpkeSuite.KDF_HKDF_SHA256;
+import static org.conscrypt.HpkeSuite.KEM_DHKEM_X25519_HKDF_SHA256;
+
+import java.security.InvalidKeyException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+
+/**
+ * Implementation of {@link HpkeSpi}.  Should not be used directly, but rather by one
+ * of the subclasses of {@link HpkeContext}.
+ */
+public class HpkeImpl implements HpkeSpi {
+  private final HpkeSuite hpkeSuite;
+
+  private NativeRef.EVP_HPKE_CTX ctx;
+  private byte[] encapsulated = null;
+
+
+  public HpkeImpl(HpkeSuite hpkeSuite) {
+    this.hpkeSuite = hpkeSuite;
+  }
+
+  @Override
+  public void engineInitSender(int mode, PublicKey publicKey, byte[] info, byte[] sKe)
+          throws InvalidKeyException {
+    checkNotInitialised();
+    if (publicKey == null) {
+      throw new InvalidKeyException("null key");
```

Comment:
The null condition is already validated in `validatePublicKeyTypeAndGetRawKey`. Shall we remove this condition and update `validatePublicKeyTypeAndGetRawKey` to throw an InvalidKeyException instead?

Reply:
Yup, the should be consistent with each other... Done!